integration_tests/test_nft_transfer.py

```diff
@@ -0,0 +1,511 @@
+            "class-hash",
+            "nft/" + dst_channel + "/nft/" + mid_src_channel + "/" + denomid,
+            home=cli_dst.data_dir,
+            node=cli_dst.node_rpc,
+            output="json",
+        )
+    )["hash"]
+
+    # get class trace on destination chain
+    dst_class_trace = json.loads(
+        cli_dst.raw(
+            "query",
+            "nft-transfer",
+            "class-trace",
+            dst_class_hash,
+            home=cli_dst.data_dir,
+            node=cli_dst.node_rpc,
+            output="json",
+        )
+    )["class_trace"]
+
+    assert dst_class_trace["base_class_id"] == denomid, dst_class_trace
+    assert (
+        dst_class_trace["path"] == "nft/" + dst_channel + "/nft/" + mid_src_channel
+    ), dst_class_trace
+
+    dst_denom_id = "ibc/" + dst_class_hash
+
+    # query denom on destination chain
+    rsp = json.loads(
+        cli_dst.raw(
+            "query",
+            "nft",
+            "denom",
+            dst_denom_id,
+            home=cli_dst.data_dir,
+            node=cli_dst.node_rpc,
+            output="json",
+        )
+    )
+
+    assert rsp["uri"] == denomuri, rsp["uri"]
+
+    # query nft on destination chain
+    rsp = json.loads(
+        cli_dst.raw(
+            "query",
+            "nft",
+            "token",
+            dst_denom_id,
+            tokenid,
+            home=cli_dst.data_dir,
+            node=cli_dst.node_rpc,
+            output="json",
+        )
+    )
+
+    assert rsp["uri"] == tokenuri, rsp
+    assert rsp["owner"] == addr_dst, rsp
+
+    # quert nft on mid chain's escrow address
+    mid_escrow_address = str(
+        cli_mid.raw(
+            "query",
+            "nft-transfer",
+            "escrow-address",
+            "nft",
+            mid_dst_channel,
+            home=cli_mid.data_dir,
+            node=cli_mid.node_rpc,
+            output="json",
+        ),
+        "UTF-8",
+    ).strip()
+
+    rsp = json.loads(
+        cli_mid.raw(
+            "query",
+            "nft",
+            "token",
+            mid_denom_id,
+            tokenid,
+            home=cli_mid.data_dir,
+            node=cli_mid.node_rpc,
+            output="json",
+        )
+    )
+
+    assert rsp["uri"] == tokenuri, rsp
+    assert rsp["owner"] == mid_escrow_address, rsp
+
+    # transfer nft back to mid chain
+    rsp = json.loads(
+        cli_dst.raw(
+            "tx",
+            "nft-transfer",
+            "transfer",
+            "nft",
+            dst_channel,
+            addr_mid,
+            dst_denom_id,
+            tokenid,
+            "-y",
+            home=cli_dst.data_dir,
+            from_=addr_dst,
+            keyring_backend="test",
+            chain_id=cli_dst.chain_id,
+            node=cli_dst.node_rpc,
+        )
+    )
+
+    assert rsp["code"] == 0, rsp["raw_log"]
+
+    # FIXME more stable way to wait for relaying
+    time.sleep(20)
+
+    # TODO: nft should be burnt on destination chain
```

Comment:
this is todo for extra assertions?

Reply:
Forgot to remove todo from comments. Will remove.

Reply:
Done